src/storage/arena_allocator.cpp

```diff
@@ -17,7 +20,36 @@ ArenaChunk::~ArenaChunk() {
 	}
 }
 
-ArenaAllocator::ArenaAllocator(Allocator &allocator, idx_t initial_capacity) : allocator(allocator) {
+//===--------------------------------------------------------------------===//
+// Allocator Wrapper
+//===--------------------------------------------------------------------===//
+struct ArenaAllocatorData : public PrivateAllocatorData {
+	explicit ArenaAllocatorData(ArenaAllocator &allocator) : allocator(allocator) {
+	}
+
+	ArenaAllocator &allocator;
+};
+
+static data_ptr_t ArenaAllocatorAllocate(PrivateAllocatorData *private_data, idx_t size) {
+	auto &allocator_data = (ArenaAllocatorData &)*private_data;
+	return allocator_data.allocator.Allocate(size);
+}
+
+static void ArenaAllocatorFree(PrivateAllocatorData *, data_ptr_t, idx_t) {
+	// nop
```

Comment:
So we do not actually free any data when using the arena allocator, right? Maybe we should make that a bit more clear in the `Destroy` functions of the list segments..?

Reply:
We don't, but the `Destroy` is orthogonal to this since the allocator is a parameter (we can still use `List` with a non-arena allocator).